python/lsst/ctrl/mpexec/simple_pipeline_executor.py

```diff
@@ -0,0 +1,298 @@
+    features present in the command-line-only ``pipetask`` tool in order to
+    keep the implementation simple.  Python callers that need more
+    sophistication should call lower-level tools like `GraphBuilder`,
+    `PreExecInit`, and `SingleQuantumExecutor` directly.
+    """
+
+    def __init__(self, quantum_graph: QuantumGraph, butler: Butler):
+        self.quantum_graph = quantum_graph
+        self.butler = butler
+
+    @classmethod
+    def prep_butler(
+        cls,
+        root: str,
+        inputs: Iterable[str],
+        output: str,
+        output_run: Optional[str] = None,
+    ) -> Butler:
+        """Helper method for creating `Butler` instances with collections
+        appropriate for processing.
+
+        Parameters
+        ----------
+        root : `str`
+            Root of the butler data repository; must already exist, with all
+            necessary input data.
+        inputs : `Iterable` [ `str` ]
+            Collections to search for all input datasets, in search order.
+        output : `str`
+            Name of a new output `~CollectionType.CHAINED` collection to create
+            that will combine both inputs and outputs.
+        output_run : `str`, optional
+            Name of the output `~CollectionType.RUN` that will directly hold
+            all output datasets.  If not provided, a name will be created from
+            ``output`` and a timestamp.
+
+        Returns
+        -------
+        butler : `Butler`
+            Butler client instance compatible with all `classmethod` factories.
+            Always writeable.
+        """
+        if output_run is None:
+            output_run = f"{output}/{Instrument.makeCollectionTimestamp()}"
+        # Make initial butler with no collections, since we haven't created
+        # them yet.
+        butler = Butler(root, writeable=True)
+        butler.registry.registerCollection(output_run, CollectionType.RUN)
+        butler.registry.registerCollection(output, CollectionType.CHAINED)
+        collections = list(inputs)
+        collections.append(output_run)
+        butler.registry.setCollectionChain(output, collections)
+        # Remake butler to let it infer default data IDs from collections, now
+        # that those collections exist.
+        return Butler(butler=butler, collections=[output], run=output_run)
+
+    @classmethod
+    def from_pipeline_filename(
+        cls, pipeline_filename: str, *, where: str = "", butler: Butler
+    ) -> SimplePipelineExecutor:
+        """Create an executor by building a QuantumGraph from an on-disk
+        pipeline YAML file.
+
+        Parameters
+        ----------
+        pipeline_filename : `str`
+            Name of the YAML file to load the pipeline definition from.
+        where : `str`, optional
+            Data ID query expression that constraints the quanta generated.
+        butler : `Butler`
+            Butler that manages all I/O.
```

Comment:
I think this docstring should mention that you can get an appropriate butler from `prep_butler`.